scripts/ledger_network_update.py

```diff
@@ -0,0 +1,398 @@
+            )
+            content = content.replace(
+                f"| RPC Endpoint   | {self.cur_config.rpc_api_address}",
+                f"| RPC Endpoint   | {self.new_config.rpc_api_address}",
+            )
+
+            content = content.replace(
+                f"| REST Endpoint  | {self.cur_config.rest_api_address}",
+                f"| REST Endpoint  | {self.new_config.rest_api_address}",
+            )
+            f.write_text(content)
+
+    def update_conftest(self):
+        """Update tests/conftest.py."""
+        click.echo("Update tests/conftest.py")
+        f = ROOT_DIR / "tests/conftest.py"
+        content = f.read_text()
+        content = content.replace(
+            f'DEFAULT_FETCH_ADDR_REMOTE = "{self.cur_config.rest_api_address}"',
+            f'DEFAULT_FETCH_ADDR_REMOTE = "{self.new_config.rest_api_address}"',
+        )
+        content = content.replace(
+            f'DEFAULT_FETCH_CHAIN_ID = "{self.cur_config.chain_id}"',
+            f'DEFAULT_FETCH_CHAIN_ID = "{self.new_config.chain_id}"',
+        )
+        f.write_text(content)
+
+    def update_packages(self):
+        """Update packages."""
+        click.echo("Update packages")
+        configs_files = (ROOT_DIR / "packages").glob("**/*.yaml")
+        for f in configs_files:
+            content = f.read_text()
+            content = content.replace(
+                f"address: {self.cur_config.rest_api_address}",
+                f"address: {self.new_config.rest_api_address}",
+            )
+            content = content.replace(
+                f"chain_id: {self.cur_config.chain_id}",
+                f"chain_id: {self.new_config.chain_id}",
+            )
+            f.write_text(content)
+
+    def update_plugins(self):
+        """Update plugins."""
+        click.echo("Update plugins")
+        files = (ROOT_DIR / "plugins").glob("**/*.py")
+        for f in files:
+            content = f.read_text()
+            content = content.replace(
+                f'DEFAULT_CHAIN_ID = "{self.cur_config.chain_id}"',
+                f'DEFAULT_CHAIN_ID = "{self.new_config.chain_id}"',
+            )
+
+            content = content.replace(
+                f'DEFAULT_ADDRESS = "{self.cur_config.rest_api_address}"',
+                f'DEFAULT_ADDRESS = "{self.new_config.rest_api_address}"',
+            )
+            content = content.replace(
+                f'FETCHAI_DEFAULT_CHAIN_ID = "{self.cur_config.chain_id}"',
+                f'FETCHAI_DEFAULT_CHAIN_ID = "{self.new_config.chain_id}"',
+            )
+
+            content = content.replace(
+                f'FETCHAI_DEFAULT_ADDRESS = "{self.cur_config.rest_api_address}"',
+                f'FETCHAI_DEFAULT_ADDRESS = "{self.new_config.rest_api_address}"',
+            )
+
+            content = content.replace(
+                f'FETCHAI_TESTNET_FAUCET_URL = "https://faucet-{self.cur_config.net_name}.t-v2-london-c.fetch-ai.com"',
```

Comment:
Strictly speaking, this element of the URL is the kubernetes namespace, not the net_name.  So far, all testnets happen to have been deployed into matching namespaces, but that's not necessarily guaranteed (and is guaranteed NOT to be the case for mainnets... not that they have faucets!).

If this update script is targetted at external users, maybe I should create, and this script should reference, faucet-{net_name}.fetch.ai CNAMEs (as we already do for rpc, rest, grpc, explore, browse etc), to better decouple user-facing URLs from internal deployment details?

Reply:
Let's take the whole Faucet URL from input.

Reply:
@5A11 done